ddl/job_table.go

```diff
@@ -14,6 +14,55 @@
 
 package ddl
 
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/util/logutil"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+var (
+	addingDDLJobGeneral = "/tidb/ddl/add_ddl_job_general"
+	addingDDLJobReorg   = "/tidb/ddl/add_ddl_job_reorg"
+)
+
+func (dc *ddlCtx) insertRunningDDLJobMap(id int64) {
+	dc.runningJobs.Lock()
+	defer dc.runningJobs.Unlock()
+	dc.runningJobs.runningJobMap[id] = struct{}{}
+}
+
+func (dc *ddlCtx) deleteRunningDDLJobMap(id int64) {
+	dc.runningJobs.Lock()
+	defer dc.runningJobs.Unlock()
+	delete(dc.runningJobs.runningJobMap, id)
+}
+
+func (dc *ddlCtx) getRunningDDLJobIDs() string {
+	dc.resetRunningIDs()
+	dc.runningJobs.RLock()
+	defer dc.runningJobs.RUnlock()
+	for id := range dc.runningJobs.runningJobMap {
+		dc.runningJobIDs = append(dc.runningJobIDs, strconv.Itoa(int(id)))
+	}
+	return strings.Join(dc.runningJobIDs, ",")
+}
+
+func (dc *ddlCtx) resetRunningIDs() {
+	dc.runningJobIDs = dc.runningJobIDs[:1]
```

Comment:
Why does it need to leave an element?

Reply:
we will build a sql `.... not in ( %s )`, it must contain one element, so I left `0` in `runningJobIDs`, and there will no `0` for job id

Reply:
It should be unnecessary.